enhancements/installer/aws-lb-subnet-selection.md

```diff
@@ -0,0 +1,1147 @@
+
+## Motivation
+
+Cluster admins using AWS may have dedicated subnets for their load balancers due to
+security reasons, architecture, or infrastructure constraints. They may be installing
+a cluster into a VPC with multiple Availability Zones (AZs) containing untagged subnets, but
+they wish to restrict their cluster to a single AZ.
+
+Currently, cluster admins can configure their IngressController subnets after installation (Day 2) by configuring
+the `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.classicLoadBalancer.subnets`
+or `...aws.networkLoadBalancer.subnets` fields (see [LoadBalancer Subnet Selection for AWS](/enhancements/ingress/lb-subnet-selection-aws.md)).
+However, there is no way to explicitly configure the ControlPlaneLB subnets. Cluster admins also need a way to
+configure both ControlPlaneLB and IngressControllerLB subnets at install time (Day 1) using the install-config.
+
+Configuring subnets at install time is essential for cluster admins that want to
+ensure their load balancer subnets are configured properly from the start. Additionally,
+Service Delivery has use cases for ROSA Hosted Control Plane (HCP) that require the `default`
+IngressController and ControlPlaneLB subnets to be configured during cluster installation.
+
+### AWS Subnet Discovery
+
+AWS subnet discovery is the mechanism used by the AWS Cloud Controller Manager (CCM) to determine the Elastic Load
+Balancer (ELB) subnets created for LoadBalancer-type Services. According to [kubernetes/kubernetes#97431](https://github.com/kubernetes/kubernetes/pull/97431),
+a subnet will **NOT** be selected if:
+
+- The `kubernetes.io/cluster/<cluster-id>` tag contains the cluster ID of another cluster.
+- The load balancer is external (internet facing) and the subnet is private.
+- A subnet for the same AZ has already been chosen based on lexicographical order.
+- The subnet type is a `wavelength` or `local`.
+
+_Note: Subnet discovery, as defined here, applies only to IngressControllerLB subnets and not to ControlPlaneLB subnets.
+While the Cluster API Provider AWS ([CAPA](https://github.com/kubernetes-sigs/cluster-api-provider-aws)) does some subnet
+filtering, it relies on the list of [provided subnets](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/8f46a4d34d18cb090a4b48c9a9260cd3dffd9162/api/v1beta2/network_types.go#L340)
+from the openshift-installer. As a result, ControlPlaneLB subnets are NOT impacted by [OCPBUGS-17432](https://issues.redhat.com/browse/OCPBUGS-17432)._
+
+However, the current subnet discovery implementation has some limitations:
+
+- It selects subnets that lack the `kubernetes.io/cluster/<cluster-id>` tag ([OCPBUGS-17432](https://issues.redhat.com/browse/OCPBUGS-17432))
+- It doesn't support use cases such as [Dedicated Load Balancer Subnets User Story](#dedicated-load-balancer-subnets-user-story)
+
+The goal of this enhancement is not to completely bypass or replace subnet discovery by hardcoding the
```

Comment:
Nit: I think this belongs in the "Motivation" section.

So, presence of this config bypasses subnet discovery. What is the customer specifies subnets with all roles except for `ClusterNode'. Would that be a valid config and we will fall back to subnet discovery just for that role OR will that be considered invalid config?

Reply:
I see that this question is answered in the "Automatic vs. Manual Role Selection" section.